src/operation/insert/mod.rs

```diff
@@ -110,17 +110,15 @@ impl<'a, T: Serialize> Operation for Insert<'a, T> {
             }
             .into());
         }
-
-        let mut options = self.options.clone().unwrap_or_default();
-        options.ordered = Some(self.is_ordered());
```

Comment:
In this case I think we need to keep around these two deleted lines to ensure `ordered` is set properly.

Reply:
Not sure I understand why we need to set `ordered` on a copy of `self.options` if we'll be using `self.options` directly in the rest of this method. Should I be passing the copied `options` to `remove_empty_write_concern` and the `InsertCommand` builder instead?

Reply:
Yeah, prior to these changes the copied `options` with the updated `ordered` field was being passed to `InsertCommand`, so we'll want to continue to do that. Essentially the only change here should be calling `remove_empty_write_concern` on the copied options.

Reply:
Done in https://github.com/mongodb/mongo-rust-driver/pull/513/commits/baa9a519f1eadef8168a9a64b08ee2b1bf859aea.